src/test/java/com/fasterxml/uuid/UUIDGeneratorTest.java

```diff
@@ -393,6 +394,105 @@ public void testGenerateNameBasedUUIDNameSpaceNameAndMessageDigest()
             Arrays.equals(uuid_array, uuid_array2));
     }
     
+    /**
+     * Test of generateTimeBasedReorderedUUID() method,
+     * of class com.fasterxml.uuid.UUIDGenerator.
+     */
+    public void testGenerateTimeBasedReorderedUUID()
+    {
+        // this test will attempt to check for reasonable behavior of the
+        // generateTimeBasedUUID method
+        
+        // we need a instance to use
+        TimeBasedReorderedGenerator uuid_gen = Generators.timeBasedReorderedGenerator();
+        
+        // first check that given a number of calls to generateTimeBasedUUID,
+        // all returned UUIDs order after the last returned UUID
+        // we'll check this by generating the UUIDs into one array and sorting
+        // then in another and checking the order of the two match
+        // change the number in the array statement if you want more or less
+        // UUIDs to be generated and tested
+        UUID uuid_array[] = new UUID[SIZE_OF_TEST_ARRAY];
+        
+        // before we generate all the uuids, lets get the start time
+        long start_time = System.currentTimeMillis();
+        
+        // now create the array of uuids
+        for (int i = 0; i < uuid_array.length; i++) {
+            uuid_array[i] = uuid_gen.generate();
+        }
+        
+        // now capture the end time
+        long end_time = System.currentTimeMillis();
+        
+        // check that none of the UUIDs are null
+        checkUUIDArrayForNonNullUUIDs(uuid_array);
+
+        // check that all the uuids were correct variant and version (type-1)
+        checkUUIDArrayForCorrectVariantAndVersion(uuid_array, UUIDType.TIME_BASED_REORDERED);
+
+        // check that all the uuids were generated with correct order
+        checkUUIDArrayForCorrectOrdering(uuid_array);
+        
+        // check that all uuids were unique
+        checkUUIDArrayForUniqueness(uuid_array);
+        
+        // check that all uuids have timestamps between the start and end time
+        checkUUIDArrayForCorrectCreationTime(uuid_array, start_time, end_time);
```

Comment:
@cowtowncoder I'm a bit suspicious that this "just works" as the encoding for the time is reordered.  This test is basically copy/past from the original test for the TimeBasedGenerator.

Reply:
Agreed, should fail :)

I think that's since timestamp slicing wasn't changed in code; I will change it and ensure test breaks =)